dashboard/app/views/levels/_teacher.html.haml

```diff
@@ -64,7 +64,8 @@
                 %button.btn.btn-info= t('reference_area.teacher')
           - if @level.try(:examples).present? && current_user.try(:authorized_teacher?) # 'solutions' for applab-type levels
             - @level.examples.each_with_index do |example, index|
-              = link_to applab_project_view_projects_url(channel_id: example, host: 'studio.code.org', port: 443, protocol: :https), target: "_blank" do
+              - example_link = send("#{@level.game.app}_project_view_projects_url".to_sym, channel_id: example, host: 'studio.code.org', port: 443, protocol: :https)
+              = link_to example_link, target: "_blank" do
```

Comment:
That's terrible. Since Rails is already making it impossible to find where this string comes from by searching for the full method name, I'm fine with you constructing the method name here.